File: cotgenerator/src/main/java/com/jon/cotgenerator/GeneratorApplication.java
package com.jon.cotgenerator;

import com.jon.common.AppSpecific;
import com.jon.common.CotApplication;

public class GeneratorApplication extends CotApplication {
    @Override
    public void onCreate() {
        super.onCreate();

        /* To supply app-specific info during runtime like package name, build time/version, etc. */
        AppSpecific.setReferenceRepo(new GeneratorRepo());
    }
}

File: cotgenerator/src/main/java/com/jon/cotgenerator/GeneratorApplication.kt
package com.jon.cotgenerator

import com.jon.common.CotApplication
import com.jon.common.variants.Variant

@Suppress("unused")
class GeneratorApplication : CotApplication() {
    override fun onCreate() {
        super.onCreate()

        /* To supply app-specific info during runtime like package name, build time/version, etc. */
        Variant.setInjector(GeneratorInjector())
    }
}
